Xamarin.Forms.Platform.Android/Renderers/WebViewRenderer.cs

```diff
@@ -47,19 +50,25 @@ protected override void Dispose(bool disposing)
 			{
 				if (Element != null)
 				{
-					if (Control != null)
-						Control.StopLoading();
+					Control?.StopLoading();
+
 					ElementController.EvalRequested -= OnEvalRequested;
 					ElementController.GoBackRequested -= OnGoBackRequested;
 					ElementController.GoForwardRequested -= OnGoForwardRequested;
 
+					_webViewClient?.Dispose();
 					_webChromeClient?.Dispose();
 				}
 			}
 
 			base.Dispose(disposing);
 		}
 
+		protected virtual FormsWebViewClient GetFormsWebViewClient()
```

Comment:
I haven't looked but can this return the base class instead of FormsWebViewClient?

Reply:
I did it this way to be consistent with the FormsWebChromeClient.

Reply:
I would prefer we return base classes whenever possible, keeps as much flexibility there as we can (so long as we dont have to cast it)

Reply:
I'm split on this because while the renderer could safely treat the client as the base class, the point of opening it up is so that people can subclass the renderer and client without losing any functionality or having to copy & paste a bunch of code. Enforcing that they provide a subclass of FormsWebViewClient keeps users from shooting themselves in the foot and accidentally losing the implementation of the Navigating and Navigated events.

If we want to give them that opportunity I think GetFormsWebChromeClient() should be changed to return the base class as well.

Reply:
@jcmanke You may or may not be interested to know that I have just shot myself in the foot by _not_ deriving my custom ```WebViewClient``` from ```FormsWebViewClient``` but thanks to your comment above I didn't spend too much time figuring out why my Navigating events still weren't firing even after upgrading to the latest version of Xamarin.Forms...